src/main/java/pocketpal/frontend/parser/ParseCommand.java

```diff
@@ -0,0 +1,128 @@
+package pocketpal.frontend.parser;
+
+import pocketpal.frontend.commands.Command;
+import pocketpal.frontend.constants.MessageConstants;
+import pocketpal.frontend.constants.ParserConstants;
+import pocketpal.frontend.exceptions.InvalidArgumentsException;
+import pocketpal.frontend.exceptions.InvalidCategoryException;
+import pocketpal.frontend.exceptions.InvalidDateException;
+import pocketpal.frontend.exceptions.MissingArgumentsException;
+import pocketpal.frontend.exceptions.MissingDateException;
+import pocketpal.frontend.exceptions.UnknownOptionException;
+import pocketpal.frontend.util.CategoryUtil;
+import pocketpal.frontend.util.StringUtil;
+
+import java.util.logging.Logger;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public abstract class ParseCommand {
+    private static final Logger logger = Logger.getLogger(ParseCommand.class.getName());
+
+    public abstract Command parseArguments(String input) throws InvalidArgumentsException,
+            InvalidCategoryException, MissingArgumentsException, MissingDateException, InvalidDateException,
+            UnknownOptionException;
+
+    public String extractDetail(String input, Pattern pattern) throws MissingArgumentsException {
+        String detail = null;
+        Matcher matcher = pattern.matcher(input);
+        if (matcher.find()) {
+            String option = matcher.group(ParserConstants.OPTION_GROUP);
+            detail = matcher.group(ParserConstants.DETAIL_GROUP);
+            if (detail != null) {
+                detail = detail.trim();
+            }
+            if ((detail == null || detail.isEmpty()) && (option != null)) { //option specified with 0 arguments
```

Comment:
Perhaps these boolean statements, as well as the statements in other sections, can be refactored into variables to avoid over-complicated statements, as recommended by the [textbook](https://nus-cs2113-ay2223s2.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-complicated-expressions)